[Bug]: Multiarch images don't have metadata

### zot version

42f9f78

### Describe the bug

When I push multi-arch images (e.g. quay.io/dougrabson/freebsd-minimal:13.2) to a zot registry, the UI doesn't show any of the metadata (e.g. description, licences etc). If I take one image from the manifest and push it to a different name, the metadata shows up.

It is possible that I am missing something when building the manifests for these images - I noticed that quay.io and docker.io also don't show the image metadata. 

### To reproduce

Copy a multi-arch image using skopeo:
```
skopeo copy --all docker://quay.io/dougrabson/freebsd-minimal@sha256:bf7bb3a8247b6f1b6c88982f932d3f95e0ee9968d3a152c0f2797a672a79e372 docker://myregistry/testimage
```
Navigate to to the testimage page in the UI and observe that description and licenses are not populated

### Expected behavior

_No response_

### Screenshots

_No response_

### Additional context

_No response_

I think it works but its hard to be sure sine my local zot instance keeps freezing, possibly because of htis?
```
$ ./bin/zot-freebsd-amd64 serve /zboot/zot/config.json                                                                                                       2 ↵
{"level":"info","trivyDownloadURL":"ghcr.io/aquasecurity/trivy-db","time":"2023-08-02T15:13:11+01:00","message":"Config: using default Trivy DB download URL."}
{"level":"info","trivyJavaDownloadURL":"ghcr.io/aquasecurity/trivy-java-db","time":"2023-08-02T15:13:11+01:00","message":"Config: using default Trivy Java DB download URL."}
panic: timeout

goroutine 1 [running]:
zotregistry.io/zot/pkg/cli.newServeCmd.func1(0x1cbd32bf6c00?, {0x1cbd31f64ca0, 0x1, 0x1?})
	zotregistry.io/zot/pkg/cli/root.go:66 +0x11f
github.com/spf13/cobra.(*Command).execute(0x1cbd32bf6c00, {0x1cbd31f64c60, 0x1, 0x1})
	github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x1cbd32bf6900)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(0x1cbd300061a0?)
	github.com/spf13/cobra@v1.7.0/command.go:992 +0x19
main.main()
	zotregistry.io/zot/cmd/zot/main.go:10 +0x1e
```

--- a/pkg/cli/cve_cmd.go
+++ b/pkg/cli/cve_cmd.go
@@ -154,7 +154,7 @@ func searchCve(searchConfig searchConfig) error {
 	}
 
 	for _, searcher := range searchers {
-		// there can be CVE DB readyness issues on the server side
+		// there can be CVE DB readiness issues on the server side
 		// we need a retry mechanism for that specific type of errors
 		maxAttempts := 20
 

--- a/pkg/extensions/search/convert/annotations.go
+++ b/pkg/extensions/search/convert/annotations.go
@@ -133,3 +133,58 @@ func GetAnnotations(annotations, labels map[string]string) ImageAnnotations {
 		Authors:       authors,
 	}
 }
+
+func GetIndexAnnotations(indexAnnotations, manifestAnnotations, manifestLabels map[string]string) ImageAnnotations {
+	annotationsFromManifest := GetAnnotations(manifestAnnotations, manifestLabels)
+
+	description := GetDescription(indexAnnotations)
+	if description == "" {
+		description = annotationsFromManifest.Description
+	}
+
+	title := GetTitle(indexAnnotations)
+	if title == "" {
+		title = annotationsFromManifest.Title
+	}
+
+	documentation := GetDocumentation(indexAnnotations)
+	if documentation == "" {
+		documentation = annotationsFromManifest.Documentation
+	}
+
+	source := GetSource(indexAnnotations)
+	if source == "" {
+		source = annotationsFromManifest.Source
+	}
+
+	licenses := GetLicenses(indexAnnotations)
+	if licenses == "" {
+		licenses = annotationsFromManifest.Licenses
+	}
+
+	categories := GetCategories(indexAnnotations)
+	if categories == "" {
+		categories = annotationsFromManifest.Labels
+	}
+
+	vendor := GetVendor(indexAnnotations)
+	if vendor == "" {
+		vendor = annotationsFromManifest.Vendor
+	}
+
+	authors := GetAuthors(indexAnnotations)
+	if authors == "" {
+		authors = annotationsFromManifest.Authors
+	}
+
+	return ImageAnnotations{
+		Description:   description,
+		Title:         title,
+		Documentation: documentation,
+		Source:        source,
+		Licenses:      licenses,
+		Labels:        categories,
+		Vendor:        vendor,
+		Authors:       authors,
+	}
+}

--- a/pkg/extensions/search/convert/convert_test.go
+++ b/pkg/extensions/search/convert/convert_test.go
@@ -770,3 +770,213 @@ func TestPaginatedConvert(t *testing.T) {
 		So(pageInfo.ItemCount, ShouldEqual, 1)
 	})
 }
+
+func TestGetOneManifestAnnotations(t *testing.T) {
+	Convey("GetOneManifestAnnotations errors", t, func() {
+		manifestAnnotations, configLabels := convert.GetOneManifestAnnotations(
+			ispec.Index{Manifests: []ispec.Descriptor{
+				{Digest: "bad-manifest"}, {Digest: "dig2"},
+			}},
+			map[string]mTypes.ManifestMetadata{
+				"bad-manifest": {
+					ManifestBlob: []byte(`bad`),
+					ConfigBlob:   []byte("{}"),
+				},
+			},
+		)
+		So(manifestAnnotations, ShouldBeEmpty)
+		So(configLabels, ShouldBeEmpty)
+
+		manifestAnnotations, configLabels = convert.GetOneManifestAnnotations(
+			ispec.Index{Manifests: []ispec.Descriptor{
+				{Digest: "bad-config"},
+			}},
+			map[string]mTypes.ManifestMetadata{
+				"bad-config": {
+					ManifestBlob: []byte("{}"),
+					ConfigBlob:   []byte("bad"),
+				},
+			},
+		)
+		So(manifestAnnotations, ShouldBeEmpty)
+		So(configLabels, ShouldBeEmpty)
+	})
+
+	Convey("Test ImageIndex2ImageSummary annotations logic", t, func() {
+		ctx := context.Background()
+
+		configLabels := map[string]string{
+			ispec.AnnotationDescription:   "ConfigDescription",
+			ispec.AnnotationLicenses:      "ConfigLicenses",
+			ispec.AnnotationVendor:        "ConfigVendor",
+			ispec.AnnotationAuthors:       "ConfigAuthors",
+			ispec.AnnotationTitle:         "ConfigTitle",
+			ispec.AnnotationDocumentation: "ConfigDocumentation",
+			ispec.AnnotationSource:        "ConfigSource",
+		}
+
+		manifestAnnotations := map[string]string{
+			ispec.AnnotationDescription:   "ManifestDescription",
+			ispec.AnnotationLicenses:      "ManifestLicenses",
+			ispec.AnnotationVendor:        "ManifestVendor",
+			ispec.AnnotationAuthors:       "ManifestAuthors",
+			ispec.AnnotationTitle:         "ManifestTitle",
+			ispec.AnnotationDocumentation: "ManifestDocumentation",
+			ispec.AnnotationSource:        "ManifestSource",
+		}
+
+		indexAnnotations := map[string]string{
+			ispec.AnnotationDescription:   "IndexDescription",
+			ispec.AnnotationLicenses:      "IndexLicenses",
+			ispec.AnnotationVendor:        "IndexVendor",
+			ispec.AnnotationAuthors:       "IndexAuthors",
+			ispec.AnnotationTitle:         "IndexTitle",
+			ispec.AnnotationDocumentation: "IndexDocumentation",
+			ispec.AnnotationSource:        "IndexSource",
+		}
+
+		imageWithConfigAnnotations := test.CreateImageWith().DefaultLayers().
+			ImageConfig(ispec.Image{
+				Config: ispec.ImageConfig{
+					Labels: configLabels,
+				},
+			}).Build()
+
+		imageWithManifestAndConfigAnnotations := test.CreateImageWith().DefaultLayers().
+			ImageConfig(ispec.Image{
+				Config: ispec.ImageConfig{
+					Labels: configLabels,
+				},
+			}).Annotations(manifestAnnotations).Build()
+
+		// --------------------------------------------------------
+		indexWithAnnotations := test.CreateMultiarchWith().Images(
+			[]test.Image{imageWithManifestAndConfigAnnotations},
+		).Annotations(indexAnnotations).Build()
+
+		repoMeta, manifestMetadata, indexData := test.GetMetadataForRepos(test.Repo{
+			Name: "repo",
+			MultiArchImages: []test.RepoMultiArchImage{
+				{MultiarchImage: indexWithAnnotations, Tag: "tag"},
+			},
+		})
+
+		digest := indexWithAnnotations.Digest()
+
+		imageSummary, _, err := convert.ImageIndex2ImageSummary(ctx, "repo", "tag", digest, true, repoMeta[0],
+			indexData[digest.String()], manifestMetadata, nil)
+		So(err, ShouldBeNil)
+		So(*imageSummary.Description, ShouldResemble, "IndexDescription")
+		So(*imageSummary.Licenses, ShouldResemble, "IndexLicenses")
+		So(*imageSummary.Title, ShouldResemble, "IndexTitle")
+		So(*imageSummary.Source, ShouldResemble, "IndexSource")
+		So(*imageSummary.Documentation, ShouldResemble, "IndexDocumentation")
+		So(*imageSummary.Vendor, ShouldResemble, "IndexVendor")
+		So(*imageSummary.Authors, ShouldResemble, "IndexAuthors")
+
+		// --------------------------------------------------------
+		indexWithManifestAndConfigAnnotations := test.CreateMultiarchWith().Images(
+			[]test.Image{imageWithManifestAndConfigAnnotations, test.CreateRandomImage(), test.CreateRandomImage()},
+		).Build()
+
+		repoMeta, manifestMetadata, indexData = test.GetMetadataForRepos(test.Repo{
+			Name:            "repo",
+			MultiArchImages: []test.RepoMultiArchImage{{MultiarchImage: indexWithManifestAndConfigAnnotations}},
+		})
+		digest = indexWithManifestAndConfigAnnotations.Digest()
+
+		imageSummary, _, err = convert.ImageIndex2ImageSummary(ctx, "repo", "tag", digest,
+			true, repoMeta[0], indexData[digest.String()], manifestMetadata, nil)
+		So(err, ShouldBeNil)
+		So(*imageSummary.Description, ShouldResemble, "ManifestDescription")
+		So(*imageSummary.Licenses, ShouldResemble, "ManifestLicenses")
+		So(*imageSummary.Title, ShouldResemble, "ManifestTitle")
+		So(*imageSummary.Source, ShouldResemble, "ManifestSource")
+		So(*imageSummary.Documentation, ShouldResemble, "ManifestDocumentation")
+		So(*imageSummary.Vendor, ShouldResemble, "ManifestVendor")
+		So(*imageSummary.Authors, ShouldResemble, "ManifestAuthors")
+		// --------------------------------------------------------
+		indexWithConfigAnnotations := test.CreateMultiarchWith().Images(
+			[]test.Image{imageWithConfigAnnotations, test.CreateRandomImage(), test.CreateRandomImage()},
+		).Build()
+
+		repoMeta, manifestMetadata, indexData = test.GetMetadataForRepos(test.Repo{
+			Name:            "repo",
+			MultiArchImages: []test.RepoMultiArchImage{{MultiarchImage: indexWithConfigAnnotations, Tag: "tag"}},
+		})
+		digest = indexWithConfigAnnotations.Digest()
+
+		imageSummary, _, err = convert.ImageIndex2ImageSummary(ctx, "repo", "tag", digest,
+			true, repoMeta[0], indexData[digest.String()], manifestMetadata, nil)
+		So(err, ShouldBeNil)
+		So(*imageSummary.Description, ShouldResemble, "ConfigDescription")
+		So(*imageSummary.Licenses, ShouldResemble, "ConfigLicenses")
+		So(*imageSummary.Title, ShouldResemble, "ConfigTitle")
+		So(*imageSummary.Source, ShouldResemble, "ConfigSource")
+		So(*imageSummary.Documentation, ShouldResemble, "ConfigDocumentation")
+		So(*imageSummary.Vendor, ShouldResemble, "ConfigVendor")
+		So(*imageSummary.Authors, ShouldResemble, "ConfigAuthors")
+		//--------------------------------------------------------
+
+		indexWithMixAnnotations := test.CreateMultiarchWith().Images(
+			[]test.Image{
+				test.CreateImageWith().DefaultLayers().ImageConfig(ispec.Image{
+					Config: ispec.ImageConfig{
+						Labels: map[string]string{
+							ispec.AnnotationDescription: "ConfigDescription",
+							ispec.AnnotationLicenses:    "ConfigLicenses",
+						},
+					},
+				}).Annotations(map[string]string{
+					ispec.AnnotationVendor:  "ManifestVendor",
+					ispec.AnnotationAuthors: "ManifestAuthors",
+				}).Build(),
+				test.CreateRandomImage(),
+				test.CreateRandomImage(),
+			},
+		).Annotations(
+			map[string]string{
+				ispec.AnnotationTitle:         "IndexTitle",
+				ispec.AnnotationDocumentation: "IndexDocumentation",
+				ispec.AnnotationSource:        "IndexSource",
+			},
+		).Build()
+
+		repoMeta, manifestMetadata, indexData = test.GetMetadataForRepos(test.Repo{
+			Name:            "repo",
+			MultiArchImages: []test.RepoMultiArchImage{{MultiarchImage: indexWithMixAnnotations, Tag: "tag"}},
+		})
+		digest = indexWithMixAnnotations.Digest()
+
+		imageSummary, _, err = convert.ImageIndex2ImageSummary(ctx, "repo", "tag", digest,
+			true, repoMeta[0], indexData[digest.String()], manifestMetadata, nil)
+		So(err, ShouldBeNil)
+		So(*imageSummary.Description, ShouldResemble, "ConfigDescription")
+		So(*imageSummary.Licenses, ShouldResemble, "ConfigLicenses")
+		So(*imageSummary.Vendor, ShouldResemble, "ManifestVendor")
+		So(*imageSummary.Authors, ShouldResemble, "ManifestAuthors")
+		So(*imageSummary.Title, ShouldResemble, "IndexTitle")
+		So(*imageSummary.Documentation, ShouldResemble, "IndexDocumentation")
+		So(*imageSummary.Source, ShouldResemble, "IndexSource")
+
+		//--------------------------------------------------------
+		indexWithNoAnnotations := test.CreateRandomMultiarch()
+
+		repoMeta, manifestMetadata, indexData = test.GetMetadataForRepos(test.Repo{
+			Name:            "repo",
+			MultiArchImages: []test.RepoMultiArchImage{{MultiarchImage: indexWithNoAnnotations, Tag: "tag"}},
+		})
+		digest = indexWithNoAnnotations.Digest()
+
+		imageSummary, _, err = convert.ImageIndex2ImageSummary(ctx, "repo", "tag", digest,
+			true, repoMeta[0], indexData[digest.String()], manifestMetadata, nil)
+		So(err, ShouldBeNil)
+		So(*imageSummary.Description, ShouldBeBlank)
+		So(*imageSummary.Licenses, ShouldBeBlank)
+		So(*imageSummary.Vendor, ShouldBeBlank)
+		So(*imageSummary.Authors, ShouldBeBlank)
+		So(*imageSummary.Title, ShouldBeBlank)
+		So(*imageSummary.Documentation, ShouldBeBlank)
+		So(*imageSummary.Source, ShouldBeBlank)
+	})
+}

--- a/pkg/extensions/search/convert/metadb.go
+++ b/pkg/extensions/search/convert/metadb.go
@@ -268,9 +268,10 @@ func ImageIndex2ImageSummary(ctx context.Context, repo, tag string, indexDigest 
 
 	indexSize = strconv.FormatInt(totalIndexSize, 10)
 
-	annotations := GetAnnotations(indexContent.Annotations, map[string]string{})
-
 	signaturesInfo := GetSignaturesInfo(isSigned, repoMeta, indexDigest)
+
+	manifestAnnotations, configLabels := GetOneManifestAnnotations(indexContent, manifestMetaMap)
+	annotations := GetIndexAnnotations(indexContent.Annotations, manifestAnnotations, configLabels)
 
 	indexSummary := gql_generated.ImageSummary{
 		RepoName:      &repo,
@@ -299,6 +300,30 @@ func ImageIndex2ImageSummary(ctx context.Context, repo, tag string, indexDigest 
 	}
 
 	return &indexSummary, indexBlobs, nil
+}
+
+func GetOneManifestAnnotations(indexContent ispec.Index, manifestMetaMap map[string]mTypes.ManifestMetadata,
+) (map[string]string, map[string]string) {
+	if len(manifestMetaMap) == 0 || len(indexContent.Manifests) == 0 {
+		return map[string]string{}, map[string]string{}
+	}
+
+	manifestMeta := manifestMetaMap[indexContent.Manifests[0].Digest.String()]
+
+	manifestContent := ispec.Manifest{}
+	configContent := ispec.Image{}
+
+	err := json.Unmarshal(manifestMeta.ManifestBlob, &manifestContent)
+	if err != nil {
+		return map[string]string{}, map[string]string{}
+	}
+
+	err = json.Unmarshal(manifestMeta.ConfigBlob, &configContent)
+	if err != nil {
+		return map[string]string{}, map[string]string{}
+	}
+
+	return manifestContent.Annotations, configContent.Config.Labels
 }
 
 func ImageManifest2ImageSummary(ctx context.Context, repo, tag string, digest godigest.Digest, skipCVE bool,

